src/Psalm/Config.php

```diff
@@ -633,6 +633,14 @@ public static function loadFromXMLFile(string $file_path, string $current_dir):
         return $config;
     }
 
+    /**
+     * Computes the hash to use for a cache folder from CLI flags and from the config file's xml contents
+     */
+    public function computeHash(): string
+    {
+        return sha1($this->hash . ':' . $this->level);
```

Comment:
I saw a wrong reuse of the cache between a partial analysis of a single file and a full run where errors due to lack of context on the first partial run was reported on the full run. Shouldn't we use a hash that is composer.lock + psalm.xml + command line to be safe?